File: scripts/fix-logger-imports.py
#!/usr/bin/env python3
"""
Script pour corriger les imports du logger mal insérés
Répare les cas où l'import logger a été inséré à l'intérieur d'un autre import
"""

import os
import re
from pathlib import Path

SRC_DIR = Path(__file__).parent.parent / 'src'
files_fixed = 0

def fix_imports(file_path):
    """Corrige les imports dans un fichier"""
    global files_fixed

    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()

        original_content = content

        # Pattern pour détecter l'import logger mal placé entre "import {" et les exports
        # Exemple: import {\nimport { logger } from '@/utils/logger';\n  Something,
        pattern = r"(import\s*\{\s*)\nimport\s*\{\s*logger\s*\}\s*from\s*['\"]@/utils/logger['\"]\s*;\s*\n(\s+)"

        if not re.search(pattern, content):
            return

        # Supprimer toutes les occurrences mal placées
        content = re.sub(pattern, r"\1\n\2", content)

        # Vérifier si logger est déjà importé correctement
        has_correct_import = re.search(r"^import\s*\{\s*logger\s*\}\s*from\s*['\"]@/utils/logger['\"]\s*;?\s*$", content, re.MULTILINE)

        if not has_correct_import:
            # Trouver le dernier import et insérer après
            lines = content.split('\n')
            last_import_idx = -1

            for i, line in enumerate(lines):
                if line.strip().startswith('import ') and not line.strip().startswith('import type'):
                    # Vérifier si c'est un import sur plusieurs lignes
                    if '{' in line and '}' not in line:
                        # Chercher la fermeture
                        for j in range(i+1, len(lines)):
                            if '}' in lines[j]:
                                last_import_idx = j
                                break
                    else:
                        last_import_idx = i

            if last_import_idx >= 0:
                # Insérer l'import après le dernier import
                lines.insert(last_import_idx + 1, "import { logger } from '@/utils/logger';")
                content = '\n'.join(lines)

        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            files_fixed += 1
            print(f"Fixed: {file_path.relative_to(SRC_DIR)}")

    except Exception as e:
        print(f"Error processing {file_path}: {e}")

def walk_directory(directory):
    """Parcourt récursivement un répertoire"""
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.ts', '.tsx')):
                file_path = Path(root) / file
                fix_imports(file_path)

def main():
    print('\nCorrection des imports logger mal places\n')
    walk_directory(SRC_DIR)
    print(f'\n{files_fixed} fichier(s) corrige(s)\n')

if __name__ == '__main__':
    main()

File: scripts/fix-logger-imports.mjs
#!/usr/bin/env node

/**
 * Script pour corriger les imports du logger mal insérés
 * Répare les cas où l'import logger a été inséré à l'intérieur d'un autre import
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const SRC_DIR = path.join(__dirname, '..', 'src');
const DRY_RUN = process.argv.includes('--dry-run');

let filesFixed = 0;

/**
 * Corrige les imports dans un fichier
 */
function fixImports(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');

    // Pattern pour détecter l'import logger mal placé
    // Cherche les cas où "import { logger }" apparaît après "import {" sans fermeture
    const brokenPattern = /^(import\s+{[^}]*)\nimport\s+{\s*logger\s*}\s+from\s+['"]@\/utils\/logger['"];?\n/gm;

    if (!brokenPattern.test(content)) {
      return;
    }

    // Reset regex
    brokenPattern.lastIndex = 0;

    let newContent = content;
    let hasChanges = false;

    // Extraire tous les imports logger mal placés
    const loggerImportPattern = /^import\s+{\s*logger\s*}\s+from\s+['"]@\/utils\/logger['"];?\n/gm;

    // Supprimer tous les imports logger (on les réinsérera correctement)
    newContent = newContent.replace(loggerImportPattern, '');

    // Vérifier s'il y avait des imports logger
    if (content.includes("from '@/utils/logger'")) {
      hasChanges = true;

      // Trouver la position après le dernier import React/externe
      const lines = newContent.split('\n');
      let lastImportIndex = -1;

      for (let i = 0; i < lines.length; i++) {
        const line = lines[i].trim();
        if (line.startsWith('import ')) {
          lastImportIndex = i;
        } else if (lastImportIndex >= 0 && line && !line.startsWith('import')) {
          // Fin des imports
          break;
        }
      }

      // Insérer l'import logger au bon endroit
      if (lastImportIndex >= 0) {
        // Vérifier si logger n'est pas déjà là
        const hasValidLoggerImport = lines.some(line =>
          line.trim().startsWith("import { logger }") &&
          line.includes("from '@/utils/logger'")
        );

        if (!hasValidLoggerImport) {
          lines.splice(lastImportIndex + 1, 0, "import { logger } from '@/utils/logger';");
        }
      } else {
        // Pas d'import trouvé, ajouter au début
        lines.unshift("import { logger } from '@/utils/logger';");
      }

      newContent = lines.join('\n');
    }

    if (hasChanges && newContent !== content) {
      if (!DRY_RUN) {
        fs.writeFileSync(filePath, newContent, 'utf8');
      }
      filesFixed++;
      console.log(`✓ Fixed: ${path.relative(SRC_DIR, filePath)}`);
    }

  } catch (error) {
    console.error(`❌ Error processing ${filePath}:`, error.message);
  }
}

/**
 * Parcourt récursivement un répertoire
 */
function walkDirectory(dir) {
  const files = fs.readdirSync(dir);

  for (const file of files) {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);

    if (stat.isDirectory()) {
      walkDirectory(filePath);
    } else if (filePath.endsWith('.ts') || filePath.endsWith('.tsx')) {
      fixImports(filePath);
    }
  }
}

/**
 * Fonction principale
 */
function main() {
  console.log('\n🔧 Correction des imports logger mal placés\n');

  if (DRY_RUN) {
    console.log('⚠️  Mode DRY-RUN - Aucun fichier ne sera modifié\n');
  }

  walkDirectory(SRC_DIR);

  console.log(`\n✅ ${filesFixed} fichier(s) corrigé(s)\n`);

  if (DRY_RUN && filesFixed > 0) {
    console.log('💡 Pour appliquer les corrections, exécutez sans --dry-run\n');
  }
}

main();
